src/Components/Components/src/DefaultComponentActivator.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Default implementation of <see cref="IComponentActivator"/>.
+    /// </summary>
+    public class DefaultComponentActivator : IComponentActivator
```

Comment:
CR note: this is `public` because people implementing a custom `IComponentActivator` may want to override the behavior in some cases, but fall back on the default behavior in other cases. We know the default behavior is just `Activator.CreateInstance`, but external developers don't necessarily know that, plus maybe we refine this behavior in some ways in the future.

Reply:
If this is public we might as well get rid of the interface.

Reply:
Can we use an abstract base type + `Default` pattern:

```c#
public abstract class ComponentActivator
{
   public static ComponentActivator Default { get; } 
   public IComponent CreateInstance();
}
```

Making the implementation makes it harder to change it (for instance requires API reviews for an additional ctor etc)

Reply:
I’d rather not have any public static instance available, because then we’re locked into never having any state that you wouldn’t want to share across users. We can do static now, but we don’t know that will always be the case.

Dropping the interface and having a base class would be OK enough I think. So too  would just be keeping the interface and not exposing any concrete implementation publicly.

Do you have preferences among those options?

Reply:
We generally preference interfaces, so let's start with that. It's pretty trivial for users to replicate our code if they absolutely need to by copying it.

Reply:
OK, done